src/python/pants/jvm/subsystems.py

```diff
@@ -48,6 +48,13 @@ def register_options(cls, register):
                 "The name must be defined as a resolve in `[jvm].resolves`.",
             ),
         )
+        register(
+            "--debug-args",
+            type=list,
+            member_type=str,
+            default=["-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005"],
```

Comment:
The port happens to be the one that IntelliJ picks by default, so... 

Note that this default (the `suspend=y` part) means that `--debug` will cause the JVM to pause while it waits for a debugger to attach. Comment if you think this isn't desirable as default behavior. For example, if the user wants to `--debug` just with printf statements, they'd have to override this.

Reply:
Maybe mention the point about the port number in the help text for this option?

Reply:
Hm. I wonder if maybe the existing `--test-debug` flag should be renamed to `--foreground` (or something) then? The `pytest` and `shell` impls don't set the relevant flags to pause in the foreground like this, but this does seem like appropriate behavior.

It doesn't seem like a good idea to collide with Intellij'd default port though. And you might also consider templating the debug port from a separate option so that folks don't have to completely redefine the debug args to change the port.

Reply:
IntelliJ acts as the client, so picking the port it connects to by default is a convenience, not a collision.

Reply:
As for templating the port, there are several other knobs in that connection string that could be templated, and that seems finicky. Better to give the user complete control.

Reply:
On balance I think maybe the default should be empty - we don't actually know how the user is going to want to debug. We can suggest this value as a common example in the help (and docs)?

Reply:
There is no equivalent flag to set for python - you have to modify your code to connect to the debugger (in Python the debugger is the server, your code connects to it), so having both `--debug` and `--foreground` might be overkill. 

Since remote debugging is an "advanced" feature - you have to know what you're doing, at least, it seems reasonable to not set this by default, at first?  Worst case, you set it in pants.toml if you always want this behavior. And if even this turns out to be an inconvenience, we can at that point start thinking about formally distinguishing debugging and foregrounding.

Reply:
> On balance I think maybe the default should be empty - we don't actually know how the user is going to want to debug. We can suggest this value as a common example in the help (and docs)?

Yea, that makes sense.